### PR TITLE
feat(training): loss scalers for spectral losses and power spectrum loss

### LINKED ISSUE #725: Current loss reduction logic makes scaling over GRID dimension mandatory
In some situations, such as in the case of spectral losses (#678), the loss scaling over the GRID dimension has to be avoided as it becomes meaningless in the spectral space. However, if a user specifies a list of scalers that does not include a scaling over the GRID dimension, they will reach an error due to this:

https://github.com/ecmwf/anemoi-core/blob/dba42684794422e46b16c76f4c17a6db4859fd2c/training/src/anemoi/training/losses/base.py#L109-L114

it turns out this is related to how the reduction logic of the loss is implemented. For the GRID dimension, the values of the loss are first reduced via a sum, not a mean, and it is then assumed that the (weighted) average is implicitly computed later via the scaler. This is not ideal because in practice it makes the scaler over the GRID dimension mandatory, even when it's not needed. The current workaround is to specify a uniform scaling as suggested in the error message.

Ideally, we should use averaging directly in the reduction instead of delegating to the scaler.

### PR DESCRIPTION
**Pull request by sahahner**
This pull request introduces several important updates to the spectral loss and scaling framework. The main changes are the addition of new spectral dimension scalers, the depreciation of the `SpectralL2Loss`, introduction of a new `PowerSpectrumLoss`, and enhancements to ensure compatibility of scalers with spectral losses. Associated tests and imports are updated accordingly.

**Spectral Losses and Scalers:**
- Added new spectral dimension scalers: `SpectralDimensionScaler` and `LinearSpectralDimensionScaler`, providing uniform and linearly weighted scaling over spectral dimensions. 
- Introduced tests for the new spectral dimension scalers to verify correct behavior and formulas.

**Loss Function Updates:**
- Depercate `SpectralL2Loss`, as it is an expensive version of the MSE
- Introduce the `PowerSpectrumLoss`, which computes L2 loss on the power spectrum (energy per total wavenumber). All relevant imports, test cases, and references have been updated to use `PowerSpectrumLoss`. 

**Compatibility and Validation:**
- Added `_assert_spectral_scalers_compatible` to ensure that scalers applied on the grid dimension in spectral losses are correctly sized for the spectral domain, preventing mismatches between scaler and data shapes. 

## Description
Issue: https://github.com/ecmwf/anemoi-core/issues/967

## What problem does this change solve?
Introduce scalers for Spectral L2 losses.

## What issue or task does this change relate to?
[](https://github.com/ecmwf/anemoi-core/issues/967)

***As a contributor to the Anemoi framework, please ensure that your changes include unit tests, updates to any affected dependencies and documentation, and have been tested in a parallel setting  (i.e., with multiple GPUs). As a reviewer, you are also responsible for verifying these aspects and requesting changes if they are not adequately addressed. For guidelines about those please refer to https://anemoi.readthedocs.io/en/latest/***

By opening this pull request, I affirm that all authors agree to the [Contributor License Agreement.](https://github.com/ecmwf/codex/blob/main/Legal/contributor_license_agreement.md)

----
📚 Documentation preview 📚: https://anemoi-training--992.org.readthedocs.build/en/992/

----
📚 Documentation preview 📚: https://anemoi-graphs--992.org.readthedocs.build/en/992/

----
📚 Documentation preview 📚: https://anemoi-models--992.org.readthedocs.build/en/992/

### CODE REVIEW

File: training/src/anemoi/training/losses/spectral.py [hunk 1]
```diff
@@ -158,18 +209,20 @@ def forward(
...
-            pred_spectral = self._to_spectral_flat(pred)
-            target_spectral = self._to_spectral_flat(target)
-            diff = torch.abs(pred_spectral - target_spectral) ** 2
+        sc_pred = self.transform.forward(pred)
+        sc_target = self.transform.forward(target)
+        pred_amp = torch.sum(
+            sc_pred.real**2 + sc_pred.imag**2,
+            dim=-2,
+        )  # sum over order (M) dim to get power per wavenumber
+        target_amp = torch.sum(sc_target.real**2 + sc_target.imag**2, dim=-2)
```
Comment (frazane) [inline training/src/anemoi/training/losses/spectral.py]: This way of computing the power per wavenumber is valid for the SHT but not for the FFT. Would it make sense to move such operations on the transform classes themselves? This way, from inside the losses we can call the same methods but the implementation will be different (and correct) based on the transform.

Comment (sahahner) [reply training/src/anemoi/training/losses/spectral.py]: yes, good point. we can move this into the transform class and use an assert to check whether this is defined

**Comment by sahahner** [comment]
Putting this into draft mode as main has moved on and people use this branch for experiments. Scalers and Power Spectrum Loss moved to this PR: https://github.com/ecmwf/anemoi-core/pull/1253
